Sources/Sentry/SentryNetworkTracker.m

```diff
@@ -60,9 +60,30 @@ - (void)disable
     }
 }
 
-- (void)urlSessionTaskResume:(NSURLSessionTask *)sessionTask
+- (BOOL)addHeadersForRequestWithURL:(NSURL *)URL
 {
+    for (id targetCheck in SentrySDK.options.tracePropagationTargets) {
+        if ([targetCheck isKindOfClass:[NSRegularExpression class]]) {
+            NSString *string = URL.absoluteString;
+            NSUInteger numberOfMatches =
+                [targetCheck numberOfMatchesInString:string
+                                             options:0
+                                               range:NSMakeRange(0, [string length])];
+            if (numberOfMatches > 0) {
+                return YES;
+            }
+        } else if ([targetCheck isKindOfClass:[NSString class]]) {
+            if ([URL.host isEqualToString:targetCheck]) {
```

Comment:
h: Every other SDK uses contains and compare with the url not only the host.
We should align this.
```suggestion
            if ([URL.absoluteString containsString:targetCheck]) {
```

Reply:
That really doesn't make sense, and if other SDKs use this, they should fix it. See my comment in the PR about this enabling very weird things like sentry.io.evil.com getting headers when you check for sentry.io.

Not that these headers are very useful for evil hackers but I mean that's what this feature is for. Checking hostnames using contains is just fundamentally flawed. I'm waiting for input from Philip and Bruno, will bring this up in the meeting tonight.

Reply:
We agreed to keep the behaviour used in the other SDKs and use `containsString` for the entire URL.